whitepapers/0011_accountant.md

```diff
@@ -0,0 +1,190 @@
+
+If the guardian receives a `Committed` status for a transfer then it can immediately proceed to signing the VAA.  If it receives an `Error` status, then it should halt further processing of the transfer (see the [Handling Rejections](#handling-rejections) section for more details).  If it receives a `Pending` status, then it should add the observation to a local database of pending transfers and wait for an event from the accountant.
+
+#### Observing Events from Wormhole Chain
+
+Each guardian sets up a watcher for the accountant watching for events emitted signalling a determination on a transfer.  An event for a transfer is emitted when there is a quorum of pre-observations.
+
+There are 2 types of events that the guardian must handle:
+
+- Transfer committed - This indicates that an observation has reached quorum and the transfer has been committed.  The guardian must verify the contents of the event and then sign the VAA if they match.  See [Threat Model](#threat-model) for more details.
+- Error - This indicates that the accountant ran into an error while processing an observation.  The guardian should halt further processing of the transfer.  See the [Handling Rejections](#handling-rejections) section for more details
+
+#### Handling stuck transfers
+
+It may be possible that some transfers don't receive enough observations to reach quorum.  This can happen if a number of guardians miss the transaction on the origin chain.  Since the accountant runs before the peer to peer gossip layer, it cannot take advantage of the existing automatic retry mechanisms that the guardian network uses.  To deal with this the accountant contract provides a `MissingObservations` query that returns a list of pending transfers for which the accountant does not have an observation from a particular guardian.
+
+Guardians are expected to periodically run this query and re-observe transactions for which the accountant is missing observations from them.
+
+#### Contract Upgrades
+
+Upgrading the contract is performed with a migrate contract governance action from the guardian network. Once quorum+1 guardians (13) have signed the migrate contract governance VAA, the [wormchain client migrate command](https://github.com/wormhole-foundation/wormhole/blob/a846036b6ebff3af6f12ff375f5c3801ada20291/wormchain/x/wormhole/client/cli/tx_wasmd.go#L148) can be ran by anyone with an authorized wallet to submit the valid governance vaa and updated wasm contract to wormchain.
+
+### Account Management
+
+The accountant keeps track of token balances on each blockchain ensuring they never become negative.  When committing a transfer, the accountant parses the tokenbridge payload to get the emitter chain, recipient chain, token chain, and token address.  These values are used to look up the relevant accounts and adjust the balances as needed.
```

Comment:
"on each blockchain" makes it sound like the accountant is running on every chain. Change to "for each blockchain"?

Reply:
Fixed in ff42eb2b1b9651e4036d6d48ae3def0570bfcd3d